static/canvas.js

```diff
@@ -318,6 +346,34 @@ class Canvas {
 
         return {x: mx, y: my};
     }
+
+    /**
+     * Changes the cursor to one of the four bidirectional resize cursors, depending on which 
+     * side or corner of a box the mouse is on.
+     * @param box the box in consideration
+     * @param mx the x-coordinate of the mouse
+     * @param my the y-coordinate of the mouse.
+     */
+    doubleArrow(box, mx, my) {
```

Comment:
Other than the simplifications I specified above, I'd recommend putting this function in the box class. It's more so related to the box than the canvas.